users/serializers.py

```diff
@@ -1,72 +1,148 @@
-    staff = serializers.BooleanField(source="is_staff", default=False)
 
-    class Meta:
-        model = UserAccount
-        fields = [
-            "id",
-            "email",
-            "username",
-            "password",
-            "active",
-            "verified",
-            "superuser",
-            "staff",
-            "created_time",
-            "last_login",
-        ]
-
-        extra_kwargs = {
-            "password": {"write_only": True},
-        }
 
+
+""" Serializers are initialised here  """
+
+class UserSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(required=True, write_only=True)
+    class Meta:
+        model = Account
+        exclude = ('id', 'is_active', 'is_superuser', 'is_staff', 'created_time', 'last_login', 'is_verified', 'groups', 'user_permissions')
+
+
+    def validate_email(self, email):
+        pattern = '[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+'
+        
+        if re.match(pattern, email):
+            if Account.objects.filter(email=email).exists():
+                raise serializers.ValidationError('Email already exists!')
+            return email
+        raise serializers.ValidationError('Invalid email format!')
+    
+    
     def create(self, validated_data):
-        validated_data["is_active"] = True
-        user = UserAccount.objects.create_user(**validated_data)
+        password = self.validated_data.get('password')
+        # pattern = r'!@#$%&\^*(),.?":{}|<>'
+        
+        error_messages = []
+        
+        # if not re.search(pattern, password):
+        #         error_messages.append(f'Passwords must contain at least one SPECIAL characters {pattern}')
+            
+        try: 
+            validate_password(password)   
+        except serializers.ValidationError as e:
+            error_messages.append(str(e))
+        
+        if error_messages:
+            raise serializers.ValidationError(error_messages)
+        
+        user = Account.objects.create_user(**validated_data)
         return user
 
-    # def update(self, instance, validated_data):
-    #     instance.email = validated_data.get('new_email', instance.email)
-    #     instance.username = validated_data.get('new_username', instance.username)
 
-    #     instance.save()
-    #     return instance 
+class ListAccountsSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Account
+        fields = '__all__'
+
 
 
 class LoginSerializer(serializers.Serializer):
     username_or_email = serializers.CharField(required=True)
     password = serializers.CharField(required=True)
 
-    class Meta:
-        fields = ["username_or_email", "password"]
+    def validate(self, data):
+        pattern = '[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+'
+        username_or_email = data.get('username_or_email')
+        password = data.get('password')
+        
+        if re.match(pattern, username_or_email):
```

Comment:
You can skip these checks and use this query I got from the `backends.py` file:
```
user = UserModel.objects.get(
    Q(username__iexact=username) | Q(email__iexact=username)
)
```

Reply:
That query is to use username or email. The checks though are for logging in a user from the form. But I’ll take a look again